src/miner.cpp

```diff
@@ -559,6 +541,86 @@ void static BitcoinMiner(CWallet *pwallet)
                     std::lock_guard<std::mutex> lock{m_cs};
                     return cancelSolver;
                 };
+
+
+#ifdef USE_TROMP_EQUIHASH
+            	//////////////////////////////////////////////////////////////////////////
+            	// TROMP EQ SOLVER START
+              	// I = the block header minus nonce and solution.
+                CEquihashInput I{*pblock};
+                CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+                ss << I;
+              	const char *header = (char *)&ss[0];
+              	unsigned int headerLen = ss.size();
+
+                // Nonce
+                // Nonce is u256 convert it to u32
+                const char *nonce = (const char*) pblock->nNonce.begin();
+
+              	// Create solver and initialize it with header and nonce.
+              	equi eq(1);
+              	eq.setnonce(header, headerLen, nonce, pblock->nNonce.size());
+
+                LogPrintf("Running TrompEquihash solver with nNonceInBlock=%s\n",
+                  		 pblock->nNonce.ToString());
+
+                // Intialization done, start algo driver.
+                eq.digit0(0);
+                eq.xfull = eq.bfull = eq.hfull = 0;
+                eq.showbsizes(0);
+                for (u32 r = 1; r < WK; r++) {
+                	r&1 ? eq.digitodd(r, 0) : eq.digiteven(r, 0);
+                	eq.xfull = eq.bfull = eq.hfull = 0;
+                	eq.showbsizes(r);
```

Comment:
I'd say that this needs at least one call to check for cancellation inside the `for` loop, but perhaps given the speed of the solver this isn't necessary (or at least, we should check what the effect is on the speed of the solver). We can probably get away with just checking for cancellation before starting the solver.